sorting/insertion_sort.cpp

```diff
@@ -1,101 +1,90 @@
 /**
- *
  * \file
- * \brief [Insertion Sort Algorithm
+ * \brief [Recursive Insertion Sort Algorithm
  * (Insertion Sort)](https://en.wikipedia.org/wiki/Insertion_sort)
  *
  * \details
  * Insertion sort is a simple sorting algorithm that builds the final
- * sorted array one at a time. It is much less efficient compared to
- * other sorting algorithms like heap sort, merge sort or quick sort.
- * However it has several advantages such as
- * 1. Easy to implement
- * 2. For small set of data it is quite efficient
- * 3. More efficient that other Quadratic complexity algorithms like
- *    Selection sort or bubble sort.
- * 4. It's stable that is it does not change the relative order of
- *    elements with equal keys
- * 5. Works on hand means it can sort the array or list as it receives.
- *
- * It is based on the same idea that people use to sort the playing cards in
- * their hands.
- * the algorithms goes in the manner that we start iterating over the array
- * of elements as soon as we find a unsorted element that is a misplaced
- * element we place it at a sorted position.
- *
- * Example execution steps:
- * 1. Suppose initially we have
- * \f{bmatrix}{4 &3 &2 &5 &1\f}
- * 2. We start traversing from 4 till we reach 1
- * when we reach at 3 we find that it is misplaced so we take 3 and place
- * it at a correct position thus the array will become
```

Comment:
this is a modification of a pre-existing algorithm. Please create a new file with the recursive suffix added to it